spec/mailers/provider_mailer_spec.rb

```diff
@@ -106,4 +106,52 @@
       expect(@mail.body.encoded).to include(provider_interface_application_choice_url(application_choice_id: @application_choice.id))
     end
   end
+
+  describe 'Send provider decision chaser email' do
+    before do
+      @course_option = course_option_for_provider_code(provider_code: 'ABC')
+      @application_choice = create(:submitted_application_choice,
+                                   course_option: @course_option,
+                                   application_form:
+                                     create(
+                                       :completed_application_form,
+                                    ))
+      @provider_user = @application_choice.provider.provider_users.first
+      @mail = mailer.chase_provider_decision(@provider_user, @application_choice)
```

Comment:
Wondering if this can be refactored along with the other `describe` blocks here.

- `@course_option = course_option_for_provider_code(provider_code: 'ABC')` seems common
- I'm not sure we need to use `create` here for `application_choice` because we are passing it into the mailer
- Same for `application_form`, I think we can use `build_stubbed(:application_form, submitted_at: <some date>, first_name: 'Blah', last_name: 'Blah')` as that's the only things we care about from the application form as far as I can see

What do you think?

Reply:
"do you even refactor bro?" 
Good shout, I moved stuff into let blocks to remove the duplication! 5bc0e19